src/archive/src/main.rs

```diff
@@ -680,8 +685,29 @@ async fn status() -> ArchiveStatus {
 /// tools, can seamlessly integrate with it. The concrete interface (method name etc.) is
 /// provisional, but works.
 #[query]
+#[candid_method(query)]
 fn __get_candid_interface_tmp_hack() -> String {
     include_str!("../archive.did").to_string()
 }
 
 fn main() {}
+
+// Order dependent: do not move above any function annotated with #[candid_method]!
+candid::export_service!();
+
+#[cfg(test)]
+mod test {
+    use crate::__export_service;
+    use candid::utils::{service_compatible, CandidSource};
+    use std::path::Path;
+
+    #[test]
+    fn check_candid_interface_compatibility() {
+        let canister_interface = __export_service();
+        service_compatible(
```

Comment:
does "compatible" mean "equal"?

Reply:
No. It is actually a subtype check i.e. the canister interface is a subtype of the `.did` service.

Reply:
Thanks. Might be worth leaving a note, from the code it's unclear if this is checking `:>` or `<:`.